src/postings/postings.rs

```diff
@@ -12,16 +12,33 @@ use crate::docset::DocSet;
 /// for merging segments or for testing.
 pub trait Postings: DocSet + 'static {
     /// The number of times the term appears in the document.
-    fn term_freq(&self) -> u32;
+    fn term_freq(&mut self) -> u32;
```

Comment:
What requires this?

Reply:
In the BitSetPostingUnion, we need to `seek` in the docset list, since we arrive at a hit from the bitset, the docsets are just used for the postitions.

```rust
impl<TDocSet: Postings> Postings for BitSetPostingUnion<TDocSet> {
    fn term_freq(&mut self) -> u32 {
        let curr_doc = self.bitset.doc();
        let mut term_freq = 0;
        for docset in &mut self.docsets {
            if docset.doc() < curr_doc {
                docset.seek(curr_doc);
            }
            if docset.doc() == curr_doc {
                term_freq += docset.term_freq();
            }
        }
        term_freq
    }
```